addons/im_livechat/models/discuss_channel.py

```diff
@@ -87,6 +95,23 @@ def _execute_command_help_message_extra(self):
     def execute_command_history(self, **kwargs):
         self.env['bus.bus']._sendone(self, 'im_livechat.history_command', {'id': self.id})
 
+
+    def toggle_allow_visitor_upload(self):
+        """ Toggle the allow_visitor_upload field of the channel and notify the
+        channel members"""
+        self.ensure_one()
+        self.allow_visitor_upload = not self.allow_visitor_upload
+        message_body = Markup("<div class='o_mail_notification o_hide_author'>%s</div>") % (
+            _("File upload is enabled") if self.allow_visitor_upload else _("File upload is disabled")
+        )
+        self.message_post(body=message_body, message_type="notification", subtype_xmlid="mail.mt_comment")
+        self.env["bus.bus"]._sendone(self, "mail.record/insert", {
+            "Channel": {
+                "id": self.id,
+                "allow_visitor_upload": self.allow_visitor_upload,
+            }
+        })
```

Comment:
Would have been best to send `Thread` notification rather than `Channel`, `channel` field on `Thread` model is deprecated :/

Reply:
Updated to use thread instead of channel :+1: